chore: Add docs for ssh tunneling (#23131)
apache/superset: docs/docs/installation/setup-ssh-tunneling.mdx

---
title: Setup SSH Tunneling
hide_title: true
sidebar_position: 13
version: 1
---

## SSH Tunneling

1. Turn on feature flag

>     - Change (this value)[https://github.com/apache/superset/blob/master/superset/config.py#L489] to `True`

Reviewer: This link is going to be broken very quickly. You can use a link to a specific revision:

https://github.com/apache/superset/blob/eb8386e3f0647df6d1bbde8b42073850796cc16f/superset/config.py#L489

But even better would be also including what feature flag should be changed:

```suggestion
    - Change (`SSH_TUNNELING`)[https://github.com/apache/superset/blob/eb8386e3f0647df6d1bbde8b42073850796cc16f/superset/config.py#L489] to `True`
```

>     - If you want to add more security when establish the tunnel we allow users to overwrite the `SSHTunnelManager` class (here)[https://github.com/apache/superset/blob/master/superset/config.py#L507]

Reviewer: ```suggestion
    - If you want to add more security when establishing the tunnel we allow users to overwrite the `SSHTunnelManager` class (here)[https://github.com/apache/superset/blob/master/superset/config.py#L507]
```

Same here, link to a specific SHA like I did above.

>     - You can also set the (`SSH_TUNNEL_LOCAL_BIND_ADDRESS`)[https://github.com/apache/superset/blob/master/superset/config.py#L508] this the host address where the tunnel will be accessible on your VPC

Reviewer: Ditto.

2. Create database w/ ssh tunnel enabled

>     - With the feature flag enabled you should now see ssh tunnel toggle

Reviewer: ```suggestion
    - With the feature flag enabled you should now see ssh tunnel toggle.
```

    - Click the toggle to enables ssh tunneling and add your credentials accordingly.
        - Superset allows for 2 different type authenticaion (Basic + Private Key). These credentials should come from your service provider.

3. Verify data is flowing

>     - After goto SQL Lab, and write a query to verify data is properly flowing.

Reviewer: ```suggestion
    - Once SSH tunneling has been enabled, go to SQL Lab and write a query to verify data is properly flowing.
```
